### PR TITLE
Implement full featured workspace as onegovbox module [WIP]

### PR DESCRIPTION
**Pull request by maethu**
Global goal: Implement full reatured ftw.workspace as we know it from teamraum.
This means this package will no longer be compatible with teamraum.

- [x] Remove obsolte code, rewrite assignable_users vocab. #75
- [x] Rewrite Readme #76 
  - [x] Supported addons
  - [x] Describe extras
- [x] Include example workflow #78 
- [x] install and configure ftw.meeting #68 
- [x] Install and configure zip export #69 
- [x] install and configure ftw.tabbedview extjs + quickupload #70
- [x] Install and configure ftw.participation #72 
- [x] Install and configure ftw.notification.* #73 
- [x] Implement egov.contactdirectory as extra. #75
- [x] Install ftw.file by default #71 
- [x] Install ftw.mail by default --> NO (Because it has several Server site deps.)
- [x] Configure addable types of TabbedViewFolder to a reasonable set.
- [x] Tooltips?? I do not provide any tooltips at all. They may be implemented if needed. 
- [x] Remove smarttruncatio jquery plugin.
- [ ] TinyMCE config

### CODE REVIEW

**Comment by maethu** [comment]
@jone Do am I missing anything??

File: README.rst [hunk 1]
```diff
@@ -1,15 +1,79 @@
...
 
-- **Overview**: Shows recently modified contents within this workspace and the structure (subfolders)
-- **Documents**: Lists files recursively
-- **Events**: Lists events (see ``ftw.meeting``) and a calendar view.
+What could happen in your project you're asking?
 
-The ``@@workspaces_view`` lists all workspaces recursively.
+- Share files
+    - Multi-upload using `collective.quickupload` - You are able to directly drop files into the Workspace, the Quickupload portlet is not necessary.
+    - `ftw.file` provides a journal, versions, D'n'D file replacement, an much more.
```
Comment (jone) [inline README.rst]: `an much more` -> `and much more`

File: development.cfg [hunk 2]
```diff
@@ -2,3 +2,9 @@
 extends =
     test-plone-4.3.x.cfg
     https://raw.github.com/4teamwork/ftw-buildouts/master/plone-development.cfg
+
+find-links +=
+    http://psc.4teamwork.ch/simple
```
Comment (jone) [inline development.cfg]: Do we have private dependencies?
I'd prefer extending https://raw.githubusercontent.com/4teamwork/ftw-buildouts/master/4teamwork-psc.cfg in the `sources.cfg`, which already includes the `find-links` extension.

File: docs/HISTORY.txt [hunk 3]
```diff
@@ -5,7 +5,12 @@ Changelog
 3.1.2 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Implement example workspaces workflow.
+  [mathias.leimgruber]
+
+- Move several features from the main policy package to ``ftw.workspace``. Check
+  Readme for details. If you upgrading to this version be carefully it may enable features you don't want.
```
Comment (jone) [inline docs/HISTORY.txt]: `be careful`

File: setup.py [hunk 4]
```diff
@@ -55,24 +50,34 @@
...
-        'z3c.relationfield',
-        'collective.js.jqsmartTruncation',
+        'collective.deletepermission',
         'ftw.activity',
         'ftw.calendar',
         'ftw.colorbox',
-        'ftw.tabbedview',
+        'ftw.file',
+        'ftw.lawgiver',
+        'ftw.meeting [pdf, zipexport, calendar]',
```
Comment (jone) [inline setup.py]: `[pdf]` should probably be moved to our `pdf` extras

File: ftw/workspace/profiles/contact/metadata.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0"?>
+<metadata>
+  <dependencies>
```
Comment (jone) [inline ftw/workspace/profiles/contact/metadata.xml]: You should add a `<version>` here, or an declare an upgrade step directory. Otherwise this profile will not be upgradeable because we do not know whether it was installed.

File: ftw/workspace/profiles/contact/metadata.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+    <dependency>profile-ftw.workspace:default</dependency>
```
Comment (jone) [inline ftw/workspace/profiles/contact/metadata.xml]: I'd probably make the `contact`-profile an extension profile. I mean it may not install `ftw.workspace:default` but the user should install both.
This would allow to later install the `:contact`-profile without resetting customizations on top of `:default`-options.
Just a thought :wink:

File: ftw/workspace/profiles/default/rolemap.xml [hunk 7]
```diff
@@ -1,7 +1,10 @@
 <?xml version="1.0"?>
 <rolemap>
+  <roles>
+    <role name="Reader"/>
```
Comment (jone) [inline ftw/workspace/profiles/default/rolemap.xml]: Isn't there already a `Reader`-role? Why is this needed?

**Comment by jone** [comment]
Inviting participants does not work, any idea?
bildschirmfoto 2016-02-02 um 18 22 18

**Comment by jone** [comment]
When upgrading, CMFPlacefulWorkflow is not installed, nor are the workflows imported.
Is it by intention that there are no upgrades?

File: README.rst [hunk 8]
```diff
@@ -1,15 +1,79 @@
...
+---------------
+You may extend the Workspace by the following features.
+
+- ``izug.ticketbox``
+- ``ftw.mail``
+- ``ftw.book``
+- ``ftw.blog``
+- ``ftw.contentpage``
+- ``ftw.labels``
+- ``ftw.donwloadtoken``
```
Comment (jone) [inline README.rst]: typo

**Comment by jone** [comment]
Lawgiver / Workflows:
Currently, the `Owner`-Role is not translated in the "sharing"-tab of the workspace.
This can be fixed by [registering the `Owner`-Role utility](https://github.com/4teamwork/kapozh.workspace/commit/7844c23b9e9a61fb26d95ead3e6597a766d24a28).
Do we want to make that here or is that the job of the policy package?

File: ftw/workspace/profiles/contact/metadata.xml [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+    <dependency>profile-egov.contactdirectory:default</dependency>
```
Comment (jone) [inline ftw/workspace/profiles/contact/metadata.xml]: ftw.contact?

File: development.cfg (hunk 2 above)
Comment (maethu) [inline development.cfg]: Done... And yes we have. 

egov.contactdirectory... if you install it with the contact extra. We may fix this later...

File: setup.py (hunk 4 above)
Comment (maethu) [inline setup.py]: The zip export bases on the pdf extra. We probably cannot split it up.

**Comment by maethu** [comment]
@jone updated

File: setup.py (hunk 4 above)
Comment (maethu) [inline setup.py]: ``ftw.pdfgenerator`` is installed by default now.
